text/0000-packages-as-optional-namespaces.md

```diff
@@ -0,0 +1,241 @@
+- Feature Name: `packages_as_namespaces`
+- Start Date: (fill me in with today's date, 2022-03-09)
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+
+Languages like C++ have open namespaces where anyone can write code in any namespace.  In C++'s case, this includes the `std` namespace and is only limited by convention.  In contrast, Rust has closed namespaces which can only include code from the original namespace definition (the crate).
+
+This proposal extends Rust to have partially open namespaces by allowing crate owners to create crates like `parent::foo` that will be available as part of the crate `parent`'s namespace.  To protect the use of open namepsaces, the owners of `parent` has exclusive access to publishing crates in that namespace.
```

Comment:
```suggestion
This proposal extends Rust to have partially open namespaces by allowing crate owners to create crates like `parent::foo` that will be available as part of the crate `parent`'s namespace.  To protect the use of open namespaces, the owners of `parent` has exclusive access to publishing crates in that namespace.
```

Reply:
> the owners of `parent` has exclusive access to publishing crates in that namespace.

Should probably be "the owners of `parent` have exclusive"